integrationtest/mapper/pojo/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/testsupport/types/JavaUtilCalendarPropertyTypeDescriptor.java

```diff
@@ -0,0 +1,142 @@
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+
+public class JavaUtilCalendarPropertyTypeDescriptor extends PropertyTypeDescriptor<Calendar> {
+
+	JavaUtilCalendarPropertyTypeDescriptor() {
+		super( Calendar.class );
+	}
+
+	@Override
+	public Optional<DefaultIdentifierBridgeExpectations<Calendar>> getDefaultIdentifierBridgeExpectations() {
+		return Optional.empty();
+	}
+
+	@Override
+	public Optional<DefaultValueBridgeExpectations<Calendar, ?>> getDefaultValueBridgeExpectations() {
+		return Optional.of( new DefaultValueBridgeExpectations<Calendar, ZonedDateTime>() {
+			@Override
+			public Class<Calendar> getProjectionType() {
+				return Calendar.class;
+			}
+
+			@Override
+			public Class<ZonedDateTime> getIndexFieldJavaType() {
+				return ZonedDateTime.class;
+			}
+
+			@Override
+			public List<Calendar> getEntityPropertyValues() {
+				List<Calendar> calendars = Arrays.asList(
+						calendar( "1930-01-01T00:00:00.00Z", TimeZone.getTimeZone( "GMT+18:00" ) ),
+						calendar( "1970-01-01T00:00:00.00Z", TimeZone.getTimeZone( "Europe/Paris" ) ),
+						calendar( "1970-01-09T13:28:59.00Z", TimeZone.getTimeZone( "Europe/Paris" ) ),
+						calendar( "2017-11-06T19:19:00.54Z", TimeZone.getTimeZone( "Europe/Paris" ) ),
+						calendar( "2017-11-06T19:19:00.54Z", TimeZone.getTimeZone( "America/Chicago" ) ),
+						calendar( Long.MAX_VALUE, TimeZone.getTimeZone( "Europe/Paris" ) ),
+						calendar( Long.MAX_VALUE, TimeZone.getTimeZone( "GMT-18:00" ) )
+				);
+				return calendars;
+			}
+
+			@Override
+			public List<ZonedDateTime> getDocumentFieldValues() {
+				List<ZonedDateTime> zonedDateTimes = Arrays.asList(
+						Instant.parse( "1930-01-01T00:00:00.00Z" ).atZone( ZoneId.of( "GMT+18:00" ) ),
+						Instant.parse( "1970-01-01T00:00:00.00Z" ).atZone( ZoneId.of( "Europe/Paris" ) ),
+						Instant.parse( "1970-01-09T13:28:59.00Z" ).atZone( ZoneId.of( "Europe/Paris" ) ),
+						Instant.parse( "2017-11-06T19:19:00.54Z" ).atZone( ZoneId.of( "Europe/Paris" ) ),
+						Instant.parse( "2017-11-06T19:19:00.54Z" ).atZone( ZoneId.of( "America/Chicago" ) ),
+						Instant.ofEpochMilli( Long.MAX_VALUE ).atZone( ZoneId.of( "Europe/Paris" ) ),
+						Instant.ofEpochMilli( Long.MAX_VALUE ).atZone( ZoneId.of( "GMT-18:00" ) )
+				);
+				return zonedDateTimes;
+			}
+
+			@Override
+			public Class<?> getTypeWithValueBridge1() {
+				return TypeWithValueBridge1.class;
+			}
+
+			@Override
+			public Object instantiateTypeWithValueBridge1(int identifier, Calendar propertyValue) {
+				TypeWithValueBridge1 instance = new TypeWithValueBridge1();
+				instance.id = identifier;
+				instance.myProperty = propertyValue;
+				return instance;
+			}
+
+			@Override
+			public Class<?> getTypeWithValueBridge2() {
+				return TypeWithValueBridge2.class;
+			}
+		} );
+	}
+
+	private static Calendar calendar(String toParse, TimeZone timeZone) {
+		return calendar( Instant.parse( toParse ).toEpochMilli(), timeZone );
```

Comment:
You're right. It seems that `LocalDateTime.parse( toParse ).atZone( zoneId ).toEpochMilli()` does not exist, may I use `LocalDateTime.parse( toParse ).atZone( zoneId ).toInstant().toEpochMilli()`? Since, you know, `GregorianCalendar#from` made a mess.

Reply:
Yep, sounds good.